### PR TITLE
rejection_reason fixed

### PR DESCRIPTION
**Pull request by war-in**
### **Please check if the PR fulfills these requirements**
- [ ] The formatter, linter, and tests all run without an error
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

## **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

It's an answer to this PR (#269)

* **What is the current behaviour?** (You can also link to an open issue here)

TransactionReceipt has the wrong data_key

## **What is the new behavior (if this is a feature change)?**

GatewayTransactionReceipt was added to distinguish between RPC and gateway transaction receipt (it has "code" field)

## **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

No

### CODE REVIEW

File: starknet_py/net/client_models.py [hunk 1]
```diff
@@ -160,6 +160,15 @@ class TransactionReceipt:
     l1_to_l2_consumed_message: Optional[L1toL2Message] = None
 
 
+@dataclass
+class GatewayTransactionReceipt(TransactionReceipt):
+    """
+    Dataclass extending TransactionReceipt with code field
+    """
+
+    code: str = None
```
Comment (ca11ab1e) [inline starknet_py/net/client_models.py:169]: FTR we were not specifically interested in the `code` value. So if you prefer to keep it simple, it's OK too :+1:

**Review by cptartur — COMMENTED** [review]
Looks good to me, please also include tests from #269 and we should be good to merge.

File: starknet_py/transaction_exceptions.py [hunk 2]
```diff
@@ -1,25 +1,17 @@
...
-        self.code = code
         self.message = message
         super().__init__(self.message)
 
     def __str__(self):
-        return (
-            f"Transaction failed with following starknet error: "
-            f"{self.code + ':' if self.code is not None else ''}{self.message}."
-        )
+        return f"Transaction failed with following starknet error: " f"{self.message}."
```
Comment (ca11ab1e) [inline starknet_py/transaction_exceptions.py:14]:
```suggestion
        return f"Transaction failed with following starknet error: {self.message}."
```

File: starknet_py/transaction_exceptions.py [hunk 3]
```diff
@@ -30,7 +22,7 @@ class TransactionRejectedError(TransactionFailedError):
     def __str__(self):
         return (
             f"Transaction was rejected with following starknet error: "
```
Comment (ca11ab1e) [inline starknet_py/transaction_exceptions.py:24]:
```suggestion
            "Transaction was rejected with following starknet error: "
```

**Review by cptartur — APPROVED** [review]

**Review by ca11ab1e — APPROVED** [review]

**Pull request merged**